recipe/recipe.yaml

```diff
@@ -37,71 +32,95 @@ source:
         - fix-argminmax.patch
 
 build:
-  number: 0
-  skip: not (match(python, python_min ~ ".*") and is_abi3)
-  python:
-    version_independent: true
+  number: ${{ build }}
 
-requirements:
-  build:
-    - if: build_platform != target_platform
-      then:
+outputs:
+  - package:
+      name: polars
+    build:
+      script: echo "No build step is needed"
+      # Since this meta package is noarch: generic, we only need to build it
+      # for one target platform (does not matter which one)
+      skip: (target_platform != 'linux-64') or not (match(python, python_min ~ ".*") and is_abi3)
+      python:
+        version_independent: true
+      noarch: generic
+      string: ${{ polars_variant | replace('-', '_') }}
+      variant:
+        down_prioritize_variant: ${{ 0 if polars_variant == 'default' else 1 }}
+    requirements:
+      run:
+        - ${{ pin_subpackage('polars-' ~ polars_variant, exact=True) }}
```

Comment:
you don't want `exact=True` here as this will pin to the build string. Use https://rattler.build/latest/reference/jinja/#the-pin_subpackage-function `lower_bound` and `upper_bound` accordingly instead.